src/macros.jl

```diff
@@ -630,6 +630,39 @@ macro with(d, body)
     esc(with_helper(d, body))
 end
 
+# TODO: New docstring for @subset
```

Comment:
What does this mean?

Reply:
The current docstring only talks about transformations, i.e. creating new columns. But `@subset` and `@orderby` don't do that, and thus need their own docstrings.

Reply:
Actually, @nalimilan I could use your advice on what to do here. 

There is a trade-off between consolidating documentation, having too-long of docstrings, and having all the information in the docstring. 

Should I write variants of this text for every macro? Or put this in the documentation `.md` file separately?

Reply:
My opinion:
* Docstring should be a complete contract for a given functionality.
* If some functionality is shared you can give a reference to a different docstring.
* Explanations of why and how should be given in a manual.

Reply:
Okay I've added a pretty lengthy discussion in the `index.md` file. 

@nalimilan @bkamins this is ready for another review.